### ISSUE TITLE
Synthetic control features

### ISSUE THREAD

**Issue by kloskas** [pymc-labs/CausalPy#252]
It doesn't seem that synthetic control can manage multiple features per unit (i.e. can't manage other than the target outcome per each unit under treatment). 

Is this functionality considered for future improvements?

**Comment by drbenvincent** [comment]
Hi. It's not clear to me if you are talking about multivariate outcome or additional predictors. Can you point me to a paper, or package, or just give a more complete description of what you mean?

**Comment by kloskas** [comment]
Yes, I was thinking about multivariate treated units. This package for example [https://github.com/OscarEngelbrektson/SyntheticControlMethods/tree/master] (https://github.com/OscarEngelbrektson/SyntheticControlMethods/tree/master) manages units with multiple features other than the target (different format: time + units are in rows, all features are in columns). 

More examples, [https://cran.r-project.org/web/packages/MSCMT/MSCMT.pdf](https://cran.r-project.org/web/packages/MSCMT/MSCMT.pdf) package in R (amongst several others in R, much more frequent in this software) that considers W weights per unit + V weights per multivariate (shared between units)

**Comment by drbenvincent** [comment]
Thanks, this clears things up a lot. I've only taken a brief look so far, but I don't see any major reason why this couldn't be done. Very initial thoughts, but I think we'd need:
* In `pymc_experiments.py` we could probably need a new `MvSyntheticControl` class which inherits from a new `MvPrePostFit` class.
* In `pymc_models.py` we'd also need a new `MvWeightedSumFitter` class.

Adding this new functionality would be cool. Can't promise anything about the timeline at this point. We'd certainly be open to pull requests. Otherwise I'm very happy to have this as an open feature request. If this is for a commercial application then it could be expedited by engaging [PyMC Labs](https://www.pymc-labs.com) in some capacity - if that is relevant, feel free to get in touch.

**Comment by drbenvincent** [comment]
I think this may be closed by #494.  Need to double check though
